File: Dao/pedidos-react-23-main/pedidos-react-23-main/src/services/brigadaService.js
import api from "./pedidosAPI";

class Brigadaservice {
  buscarTodas() {
    return api.get("/brigadas");
  }
  buscarPorDescripcion(descripcion) {
    return api.get(`/brigadas?descripcion=${descripcion}`);
  }
  buscarPorId(id) {
    return api.get(`/brigadas/${id}`);
  }  
  buscarPorNombre(nombre) {
    return api.get(`/brigadas?nombre=${nombre}`);
  }
  crear(data) {
    return api.post("/brigadas", data);
  }
  modificar(id, data) {
    return api.put(`/brigadas/${id}`, data);
  }
  eliminar(id) {
    return api.delete(`/brigadas/${id}`);
  }
 
}

const service = new Brigadaservice();
export default service 

File: Dao/pedidos-rest-24-main/pedidos-rest-24-main/src/main/java/es/uvigo/mei/pedidos/servicios/BrigadaService.java
package es.uvigo.mei.pedidos.servicios;

import java.util.List;
import java.util.Optional;

import es.uvigo.mei.pedidos.entidades.Brigada;




public interface BrigadaService {
    public Brigada crear(Brigada brigada);
	public Brigada modificar(Brigada brigada);
	public void eliminar(Brigada brigada);
	
	public List<Brigada> buscarTodos();
	public List<Brigada> buscarPorNombre(String nombre);
	public Optional<Brigada> buscarPorId(long  id);
	public List<Brigada> buscarPorPatronDescripcion(String patron);

	

	
	
}
